src/metadata.ts

```diff
@@ -15,6 +15,13 @@ export type Metadata = {
 export function normalizeMetadata(input?: File | Response | BufferLike | StreamLike, encodedName?: any, size?: number | bigint): Metadata {
   if (encodedName !== undefined && !(encodedName instanceof Uint8Array)) encodedName = encodeString(encodedName)
 
+  // Ensure that if it's a directory, name ends with `/`.
+  if (encodedName && !input && size === undefined) {
```

Comment:
The input must be undefined, not merely falsy. An empty string should create an empty file, not a folder.